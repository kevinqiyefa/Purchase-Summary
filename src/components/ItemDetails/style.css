.Item-Details-Main {
  width: 15em;
}

#add::after {
  content: '\002B';
}

#minus::after {
  content: '\2212';
}

.Item-Details-Title {
  display: flex;
  align-items: center;
  cursor: pointer;
}

.Item-Details-Main h4 {
  text-decoration: underline;
}

#add,
#minus {
  padding-left: 0.5em;
  font-size: 2em;
  padding-bottom: 0.25em;
}

.item-details {
  display: flex;
  margin-bottom: 1em;
}

#item-name {
  font-size: 0.8em;
}

#item-qty {
  display: flex;
  justify-content: space-between;
  font-size: 0.9em;
  margin-bottom: 0.5em;
}

#item-picture img {
  width: 5em;
  padding-top: 0.5em;
}
.item-saving-prices {
  display: flex;
  flex-direction: column;
}

.item-subtotal {
  color: gray;
  text-decoration: line-through;
}
