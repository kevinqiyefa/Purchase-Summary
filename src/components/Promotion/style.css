.Promotion-Section-Title {
  display: flex;
  align-items: center;
  cursor: pointer;
}

.Promotion-Main h4 {
  text-decoration: underline;
  margin: 0.8em 0;
}

.promotion-form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
}

.promotion-form > div {
  display: flex;
  justify-content: space-between;
}

#promotion-form-input-label {
  color: rgba(0, 0, 0, 0.6);
  font-size: 0.9em;
  padding-left: 0.5em;
  font-weight: 500;
}

#promotion-form-input-field {
  margin-right: 2em;
  border: 1px solid gray;
}

#promotion-form-button {
  border: 1px solid rgb(104, 104, 104);
  color: black;
  padding: 2px 8px 5px 8px;
  font-weight: 700;
  cursor: pointer;
  border-radius: 25px;
}

#promotion-form-button:hover {
  background: #2196f3;
  border: 1px solid #2196f3;
  color: white;
}
.promotion-form span {
  color: red;
  font-size: 0.7em;
}
