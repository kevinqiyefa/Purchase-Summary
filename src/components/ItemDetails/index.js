import React, { Component, Fragment } from 'react';
import './style.css';
import itemImage from '../../assets/red-chair.jpeg';

class ItemDetails extends Component {
  state = {
    isActive: false
  };

  handleClick = () => {
    this.setState({ isActive: !this.state.isActive });
  };
  render() {
    const { item_name, quantity } = this.props.itemDetails;
    const { savings, subtotal } = this.props.pricing;
    const discountedPrice = (subtotal - savings).toFixed(2);

    const item = (
      <div className="item-details">
        <div id="item-picture">
          <img src={itemImage} alt="chair" />
        </div>
        <div>
          <div id="item-name">{item_name}</div>
          <div id="item-qty">
            <div className="item-saving-prices">
              <span className="bold-text">${discountedPrice}</span>
              <span className="bold-text item-subtotal">${subtotal}</span>
            </div>
            <div>Qty: {quantity}</div>
          </div>
        </div>
      </div>
    );

    const itemDetails = this.state.isActive ? (
      <Fragment>
        <div className="Item-Details-Title" onClick={this.handleClick}>
          <h4>Hide item details </h4> <span className="minus" />
        </div>
        {item}
        <hr />
      </Fragment>
    ) : (
      <div onClick={this.handleClick} className="Item-Details-Title">
        <h4>See item details </h4> <span className="add" />
      </div>
    );
    return <div className="Item-Details-Main">{itemDetails}</div>;
  }
}

export default ItemDetails;
