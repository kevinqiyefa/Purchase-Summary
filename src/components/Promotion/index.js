import React, { Component } from 'react';
import './style.css';

class Promotion extends Component {
  state = {
    showForm: false,
    promoCode: ''
  };

  handleClick = () => {
    this.setState({ showForm: !this.state.showForm });
  };

  handlePromoCodeChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleApplyPromoCode = () => {
    //you can only apply the promo code once
    //should add a temp state to check when should display error messages
    if (
      this.state.promoCode === this.props.promoCode &&
      !this.props.discountApplied
    ) {
      this.setState({ promoCode: '', showForm: false });

      this.props.applyPromoCode();
    } else {
      this.setState({ promoCode: '' });
    }
  };

  render() {
    const errorMessage = this.props.discountApplied ? (
      <span>You can only apply once</span>
    ) : null;

    const promotionForm = (
      <form className="promotion-form">
        <label htmlFor="input-promo-code" id="promotion-form-input-label">
          Promo code
        </label>
        <div>
          <input
            type="text"
            name="promoCode"
            id="promotion-form-input-field"
            onChange={this.handlePromoCodeChange}
            value={this.state.promoCode}
          />
          <input
            type="button"
            value="Apply"
            id="promotion-form-button"
            onClick={this.handleApplyPromoCode}
          />
        </div>
        {errorMessage}
      </form>
    );

    let promoSection = this.state.showForm ? (
      <div>
        <div onClick={this.handleClick} className="Promotion-Section-Title">
          <h4>Hide promo code </h4> <span className="minus" />
        </div>

        {promotionForm}
      </div>
    ) : (
      <div onClick={this.handleClick} className="Promotion-Section-Title">
        <h4>Apply promo code </h4> <span className="add" />
      </div>
    );
    return <div className="Promotion-Main">{promoSection}</div>;
  }
}

export default Promotion;
