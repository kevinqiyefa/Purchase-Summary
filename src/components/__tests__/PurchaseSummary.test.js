import React from 'react';
import { shallow } from 'enzyme';
import PuchaseSummary from '../PurchaseSummary';
import PricingSummary from '../PriceSummary';
import ItemDetails from '../ItemDetails';
import Promotion from '../Promotion';
import toJson from 'enzyme-to-json';

describe('Testing PuchaseSummary Component', () => {
  let wrapped;
  const pricingData = {
    pricing: {
      subtotal: 102.96,
      savings: 3.85,
      tax: 8.92,
      total: 108.03,
      zip: 85050
    },
    itemDetails: {
      item_name:
        'Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red',
      quantity: 1
    },
    promo_code: 'DISCOUNT',
    discount_applied: false
  };

  beforeEach(() => {
    wrapped = shallow(<PuchaseSummary pricingData={pricingData} />);
  });

  it('shows a pricing summary section', () => {
    expect(wrapped.find(PricingSummary).length).toEqual(1);
  });

  it('shows a itemDetails section', () => {
    expect(wrapped.find(ItemDetails).length).toEqual(1);
  });

  it('shows a promotion section', () => {
    expect(wrapped.find(Promotion).length).toEqual(1);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
