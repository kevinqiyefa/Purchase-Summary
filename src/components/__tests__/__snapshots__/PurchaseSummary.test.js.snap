// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PuchaseSummary Component matches snapshot 1`] = `
<div
  className="Purchase-Summary-Main"
>
  <PriceSummary
    pricing={
      Object {
        "savings": 3.85,
        "subtotal": 102.96,
        "tax": 8.92,
        "total": 108.03,
        "zip": 85050,
      }
    }
  />
  <ItemDetails
    itemDetails={
      Object {
        "item_name": "Essntials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
        "quantity": 1,
      }
    }
    pricing={
      Object {
        "savings": 3.85,
        "subtotal": 102.96,
        "tax": 8.92,
        "total": 108.03,
        "zip": 85050,
      }
    }
  />
  <Promotion
    applyPromoCode={[Function]}
    discountApplied={false}
    promoCode="DISCOUNT"
  />
</div>
`;
